awx_collection/plugins/modules/job_template.py

```diff
@@ -575,8 +575,11 @@ def main():
             new_fields['project'] = project_data['id']
         else:
             new_fields['project'] = module.resolve_name_to_id('projects', project)
-    if webhook_credential is not None:
+
+    if webhook_credential is not None and webhook_credential != '':
```

Comment:
What happens if the user leaves off webhook credential? This seems like this would set it to `None`.

We have no set pattern for this but you have sparked @jbradberry interest.

Reply:
@kk-at-redhat I believe correct logic needs to distinguish the case "user didn't include this key-value" and "user explicitly set it to something empty/false", which only using `module.params.get(...)` with the default `None` fallback doesn't get us.  I'm thinking something like:

```
if 'webhook_credential' in module.params:
    if webhook_credential:
        new_fields['webhook_credential'] = module.resolve_name_to_id('credentials', webhook_credential)
    else:
        new_fields['webhook_credential'] = None
```

Reply:
/retest